packages/sdk/src/types/runtime/behaviors/clientAction.ts

```diff
@@ -0,0 +1,57 @@
+/*!
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { Context } from '..';
+import { SetAudioStateOptions } from '../../..';
+import * as Payloads from '../../network/payloads';
+
+// tslint:disable: max-classes-per-file
+
+export abstract class ClientAction {
+	/** @hidden */
+	public abstract toJSON(): any;
+
+	/** @hidden */
+	public sendPayload(context: Context, targetId: string, actionName: string, actionState: string) {
+		context.internal.sendPayload({
+			type: 'set-client-action',
+			clientAction: {
```

Comment:
This is the level where we would need to know the client action type to allocate the C# object for Newtonsoft to populate.

Reply:
I think you will have what you need in the deserializer. There will always be a JObject['clientAction'], and it will always have a JObject['type'].

---
In reply to: [316824978](https://github.com/microsoft/mixed-reality-extension-sdk/pull/425#discussion_r316824978) [](ancestors = 316824978)

Reply:
Yeah.  I misread the converter code.  Wrote that a while ago.